### PR TITLE
[HHH-18505] A additional check to avoid NPE

### PR DESCRIPTION
**Pull request by schernolyas**
This PR adds additional checking for NPE in method "org.hibernate.processor.util.TypeUtils.hasAnnotation"
----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

https://hibernate.atlassian.net/browse/HHH-18505

### CODE REVIEW

**Comment by gavinking** [comment]
Rather than just adding a bandaid, which is fragile, I would prefer to understand why this method is being called with a null argument.

**Comment by schernolyas** [comment]
> Rather than just adding a bandaid, which is fragile, I would prefer to understand why this method is being called with a null argument.

you are right. I will try to investigate that. but this task isn't simple.

**Comment by schernolyas** [comment]
> Rather than just adding a bandaid, which is fragile, I would prefer to understand why this method is being called with a null argument.

it is modules (Java 9). I have added the checking to avoid it.

File: tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java [hunk 1]
```diff
@@ -364,6 +365,11 @@ private void processClasses(RoundEnvironment roundEnvironment) {
 		}
 
 		for ( Element element : roundEnvironment.getRootElements() ) {
+			if (isModule( element ) ) {
+				context.logMessage( Diagnostic.Kind.WARNING,
+									"Element '" + element + "' is module. Pass it." );
+				continue;
+			}
```
Comment (gavinking) [inline tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java:372]: I don't think we want to completely exclude modules from processing.

I think this solution is better: https://github.com/hibernate/hibernate-orm/pull/8802

Comment (schernolyas) [reply tooling/metamodel-generator/src/main/java/org/hibernate/processor/HibernateProcessor.java:372]: well ... ok. let's try your solution. the main thing is that it works :-)

**Comment by gavinking** [comment]
Thanks for this @schernolyas

**Pull request closed without merging**